xbout/boutdataset.py

```diff
@@ -169,7 +169,8 @@ def to_restart(self, savepath='.', nxpe=None, nype=None,
 
     def animate_list(self, variables, animate_over='t', save_as=None, show=False, fps=10,
                      nrows=None, ncols=None, poloidal_plot=False, subplots_adjust=None,
-                     vmin=None, vmax=None, logscale=None, titles=None, **kwargs):
+                     vmin=None, vmax=None, logscale=None, titles=None, aspect='equal',
+                     tight_layout=False, **kwargs):
```

Comment:
For me when `tight_layout` is set to `False` the plots overlap. But not when set to `True`.

Reply:
Layout is a pain with matplotlib :angry:

From https://matplotlib.org/3.1.1/tutorials/intermediate/tight_layout_guide.html
> tight_layout() only considers ticklabels, axis labels, and titles. Thus, other artists may be clipped and also may overlap.

So I think `tight_layout()` is liable to make the timeline slider overlap the plots. On the other hand, without it I've often seen the axis titles overlap other axes. I'd argue for not setting `tight_layout()` by default, because I think it can override other layout settings when it's called (e.g. https://stackoverflow.com/a/19153578), and doesn't always do the right thing.

@rdoyle45 could you try `constrained_layout` https://matplotlib.org/3.1.1/tutorials/intermediate/constrainedlayout_guide.html? I've not seen it before, but it looks like `matplotlib` may be trying to fix some of the problems with `tight_layout()`

Reply:
Unfortunately `constrained_layout` does not work for me when using the time-slider and pause button.

Reply:
I can't at the moment see a way to get either `tight_layout()` or `constrained_layout` to work with `animatplot`'s time-slider and pause button. Suspect it might be necessary to change `animatplot` code, if it's possible at all.

Reply:
So this is what I get if I keep the standard options i.e. `tight_layout=False` and `controls=True`:

![no_tight_layout](https://user-images.githubusercontent.com/48983334/70942703-27293400-2047-11ea-83e1-c4dbe2b993ba.gif)

But if I reverse those options i.e. `tight_layout=True` and `controls=False` I get this:

![tight_layout](https://user-images.githubusercontent.com/48983334/70942705-27c1ca80-2047-11ea-8195-4f45a6b43988.gif)

In the first the overlapping of the timeslider is fixed but the lower title overlaps the upper x-axis. In the second this is solved but we lose the timeslider.

Is this what you get John?

Reply:
Yep, I see the same thing. I've been using `subplots_adjust={'hspace':.4}` to fix the first case. Unfortunately this method can't be a general fix because whether extra space is needed, and how much, depends on the aspect ratio of the figure and each axes instance.

Reply:
Yeah. I can't seem to find a general solution to the problem either. I think setting the initial settings to `tight_layout=True` and `controls=False` is our best bet, as at least the plots don't overlap. If people want controls, for now the bar will overlap the plots.

Reply:
Just a thought - The location of the timeslider is set as a float value between 0 and 1 representing the size of the figure if I remember correctly. Is it possible to set the "box" which contains all the subplots to be lets say the top 80% of the figure and the timeslider to be in the bottom 20%?
This might not be possible but it theoretically the subplot section would scale proportionally.

Reply:
One version that might work well enough for now is to call `tight_layout` before adding the controls. The vertical space gets squeezed a bit when the controls are added, but the default settings seem to at least make the plots not overlap. We should also allow passing arguments to tight_layout - I'll make an update.

Reply:
In principle we could mess around with the `h_pad` used by `tight_layout` to add a bit of extra space when calling `tight_layout()` if controls will be added, which will then be squashed when the controls are added. But I think the current implementation is good enough for now. It might be possible to patch animatplot to make the controls interact nicely with `tight_layout()` (modifying the gridspec of the figure to add an extra row for the controls, and then building their axes in that row). If that ever happens, we can move the `tight_layout` to after the controls are added.

Reply:
I think you have cracked it John. I have tried it with up to 9 animations with no problems with overlap anywhere.